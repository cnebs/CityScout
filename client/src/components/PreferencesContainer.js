import React from 'react';
import Preference from './Preference';
import { List } from '@material-ui/core/';
import { makeStyles } from '@material-ui/core/styles';
import Title from './Title.js';

const useStyles = makeStyles(theme => ({
  root: {
    maxHeight: '55vh',
    overflow: 'auto'
  }
}))

export default function PreferencesContainer({
  categories,
  userPreferences,
  handleChange,
  userToken
}) {
  const classes = useStyles();

  return (
    <>
      <Title>What are you into?</Title>
      <List className={classes.root}>
        {categories
          .filter(cat => cat !== 'undefined')
          .map((cat, ind) => (
            <Preference
              key={ind}
              cat={cat}
              handleChange={handleChange}
              userToken={userToken}
              userPreference={
                userPreferences
                  ? userPreferences.find(pref => pref['name'] === cat)
                  : ''
              }
            />
          ))}
      </List>
    </>
  );
}
