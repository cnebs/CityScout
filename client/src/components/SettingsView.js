import React from 'react';
import axios from 'axios';
import PreferencesContainer from './PreferencesContainer';
import UnavailableTime from './UnavailableTime';
import { Grid, Typography, Avatar } from '@material-ui/core/';

export default class SettingsView extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      categories: [],
      userPreferences: []
    };
    this.postNewPreference = this.postNewPreference.bind(this);
    this.getCategories = this.getCategories.bind(this);
  }
  componentDidMount() {
    this.getCategories();
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.props.userToken !== prevProps.userToken) {
      this.getCategories();
    }
  }

  getCategories() {
    axios
      .get(`/api/categories?token=${this.props.userToken}`)
      .then(results => results.data)
      .then(({ categories, userPreferences }) =>
        this.setState({ categories, userPreferences })
      );
  }

  postNewPreference({ category, id, token, preferred }) {
    axios
      .post(`/api/categories`, {
        category,
        id,
        token,
        preferred
      })
      .then(results =>
        this.setState({ userPreferences: results.data.userPreferences })
      );
  }
  render() {
    return (
      <>
        <Grid container spacing={1} justify="center" alignItems="flex-start">
          <Grid item>
            <Avatar
              align="center"
              alt="User Avatar"
              src={this.props.user ? this.props.user.avatar_url : ''}
              style={{ marginTop: 10, width: 40, height: 40 }}
            />
          </Grid>
          <Grid item>
            <Typography variant="h4" style={{ marginTop: 10 }} align="center">
              Hello {this.props.user ? this.props.user.first_name : ''}
            </Typography>
          </Grid>
          <Grid item xs={12}>
            <Typography variant="subtitle1" align="center">
              Help us custom tailor your CityScout experience!
            </Typography>
          </Grid>
          <Grid item xs={4}>
            <PreferencesContainer
              categories={this.state.categories}
              userPreferences={this.state.userPreferences}
              handleChange={this.postNewPreference}
              userToken={this.props.userToken}
            />
          </Grid>
          <Grid item xs={8}>
            <UnavailableTime userToken={this.props.userToken} />
          </Grid>
        </Grid>
      </>
    );
  }
}

SettingsView.propTypes = {};
