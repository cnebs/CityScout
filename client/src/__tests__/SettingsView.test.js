import { shallow } from 'enzyme';
import React from 'react';
import SettingsView from '../components/SettingsView';
import PreferencesContainer from '../components/PreferencesContainer';
import { Grid, Avatar, Typography } from '@material-ui/core/';

describe('Settings View component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<SettingsView user={{ avatar_url: 'foo' }} />);
  });
  test('it renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('it renders the corect child components', () => {
    expect(wrapper.find(PreferencesContainer).length).toBe(1);
    expect(wrapper.find(Grid).length).toBeGreaterThan(0);
    expect(wrapper.find(Typography).length).toBeGreaterThan(0);
    expect(wrapper.find(Avatar).length).toBe(1);
  });
});
