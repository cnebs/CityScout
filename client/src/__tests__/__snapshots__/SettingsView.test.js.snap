// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings View component it renders without crashing 1`] = `ShallowWrapper {}`;
